src/test/kotlin/com/cultureamp/eventsourcing/PackageRemovingEventTypeResolverTest.kt

```diff
@@ -0,0 +1,43 @@
+package com.cultureamp.eventsourcing
+
+import com.cultureamp.developmodule.eventsourcing.bar.BarAggregate
+import com.cultureamp.developmodule.eventsourcing.bar.BarEvent
+import com.cultureamp.developmodule.eventsourcing.baz.BazAggregate
+import com.cultureamp.developmodule.eventsourcing.baz.BazEvent
+import com.cultureamp.developmodule.eventsourcing.baz.FirstBazEvent
+import com.cultureamp.developmodule.eventsourcing.baz.SecondBazEvent
+import com.cultureamp.developmodule.eventsourcing.foo.FirstFooEvent
+import com.cultureamp.developmodule.eventsourcing.foo.FooAggregate
+import com.cultureamp.developmodule.eventsourcing.foo.FooEvent
+import com.cultureamp.developmodule.eventsourcing.foo.SameNameEvent
+import io.kotest.core.spec.style.DescribeSpec
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Assertions.assertEquals
+
+internal class PackageRemovingEventTypeResolverTest : DescribeSpec({
+    describe("RelationalDatabaseEventsSequenceStats") {
+        it("fails with meaningful error when same event name exists in multiple aggregates") {
+            val error = Assertions.assertThrows(IllegalArgumentException::class.java) {
+                PackageRemovingEventTypeResolver(mapOf(FooAggregate::class to FooEvent::class, BarAggregate::class to BarEvent::class))
```

Comment:
It's hard to verify this in PR because the classes are defined elsewhere. Why not just inline them in this file?

Reply:
Also the Aggregate and Event names don't indicate that they share an event. it would help to have them named to make it clear that they share an event eg `AggregateWithSharedEventA` or give them well-understood domain names eg `Bookshop` and `Store` with `AddedToCart` event in both

Reply:
I needed to simulate a way where the class names are the same, but the packages are different, and verify that an error happens regarding that clash

Reply:
I don't _think_ you can do different packages within a file, but I might be wrong there!

Reply:
Oh I see. you might be able to use nested classes, since if they're in a different class they can have the same name

Reply:
That sounds a little fiddly because it's not longer just a package definition but also the sort of class nesting as well - I'm not sure if it will behave unpredictable. But I'm also not sure how it makes things better - the classes are here in the PR so I think that makes it pretty easy to review? And it matches the precedent of these other tests that put their test class definitions in a different subpackage

Reply:
OK how about naming - it's not at all clear that Foo and Bar have a shared event without inspecting both classes? And with the classes in other files, they're liable to be picked up in other tests and get strange results

Reply:
Would you have a specific suggestion for how to name these? I think it's ok that you need to inspect the classes, but I also think that the event is named `SameNameEvent` is also the hint that this is where the name clash will be. I just can't think of a neater way to do it and I think the unit test well documents whats going on as well